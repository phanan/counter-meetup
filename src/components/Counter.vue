<template>
  <div>
    <span class="result">{{ count }}</span>
    <button @click="inc">Count</button>
  </div>
</template>

<script>
export default {
  data: () => ({
    count: 0
  }),

  methods: {
    inc () {
      this.count++
    }
  }
}
</script>
