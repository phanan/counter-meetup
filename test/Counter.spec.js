import { shallow, renderToString } from '@vue/test-utils'
import Counter from '@/components/Counter.vue'

describe('Counter.vue', () => {
  it('renders', () => {
    const wrapper = shallow(Counter)
    expect(wrapper.contains('.result')).toBe(true)
    expect(wrapper.contains('button')).toBe(true)
  })

  it('counts', () => {
    const wrapper = shallow(Counter)
    expect(wrapper.find('.result').text()).toBe('0')
    wrapper.find('button').trigger('click')
    expect(wrapper.find('.result').text()).toBe('1')
  })
})
