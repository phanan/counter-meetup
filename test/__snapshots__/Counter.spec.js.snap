// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.vue matches snapshot 1`] = `<div><span class="result">0</span> <button>Count</button></div>`;
